src/energyplus/ForwardTranslator/ForwardTranslateThermalZone.cpp

```diff
@@ -885,8 +877,15 @@ namespace energyplus {
               // point the sizing object to the outdoor air spec
               sizingZoneIdf->setString(Sizing_ZoneFields::DesignSpecificationOutdoorAirObjectName, designSpecificationOutdoorAir->nameString());
             } else {
+              if (needToRegisterDSOAList) {
+                m_idfObjects.emplace_back(dsoaList.get());
+                sizingZoneIdf->setString(Sizing_ZoneFields::DesignSpecificationOutdoorAirObjectName, dsoaList->nameString());
+                needToRegisterDSOAList = false;
+              }
+
               // push an extensible group on the DSOA:SpaceList
               dsoaList->pushExtensibleGroup(std::vector<std::string>{space.nameString(), thisDSOA->nameString()});
+              atLeastOneDSOAWasWritten = true;
```

Comment:
Register DSOAList if at least one was found. And no need to write the empty DSOA in that case